src/hash_to_curve.rs

```diff
@@ -165,14 +170,152 @@ where
         let mut us = [C::Base::ZERO; 2];
         hash_to_field("SSWU", curve_id, domain_prefix, message, &mut us);
 
-        let [q0, q1]: [C; 2] = us.map(|u| simple_svdw_map_to_curve(u, z));
+        let [q0, q1]: [C; 2] = us.map(|u| simple_svdw_map_to_curve::<C>(u, z));
 
         let r = q0 + &q1;
         debug_assert!(bool::from(r.is_on_curve()));
         r
     })
 }
 
+// Implementation of <https://datatracker.ietf.org/doc/html/rfc9380#name-simplified-swu-for-ab-0>
+#[allow(clippy::type_complexity)]
+pub(crate) fn simple_svdw_hash_to_curve_secp256k1<'a>(
+    _curve_id: &'static str,
+    domain_prefix: &'a str,
+) -> Box<dyn Fn(&[u8]) -> Secp256k1 + 'a> {
+    Box::new(move |message| {
+        let rp = IsoSecp256k1::hash_to_curve(domain_prefix)(message);
+
+        let r = iso_map_secp256k1(rp);
+
+        debug_assert!(bool::from(r.is_on_curve()));
+        r
+    })
+}
+
+/// 3-Isogeny Map for Secp256k1
+/// Reference: <https://www.rfc-editor.org/rfc/rfc9380.html#name-3-isogeny-map-for-secp256k1>
+pub fn iso_map_secp256k1(rp: IsoSecp256k1) -> Secp256k1 {
+    // constants for secp256k1 iso_map computation
+    const K: [[&str; 4]; 5] = [
+        ["0x00", "0x00", "0x00", "0x00"],
+        [
+            "0x8e38e38e38e38e38e38e38e38e38e38e38e38e38e38e38e38e38e38daaaaa8c7",
+            "0x7d3d4c80bc321d5b9f315cea7fd44c5d595d2fc0bf63b92dfff1044f17c6581",
+            "0x534c328d23f234e6e2a413deca25caece4506144037c40314ecbd0b53d9dd262",
+            "0x8e38e38e38e38e38e38e38e38e38e38e38e38e38e38e38e38e38e38daaaaa88c",
+        ],
+        [
+            "0xd35771193d94918a9ca34ccbb7b640dd86cd409542f8487d9fe6b745781eb49b",
+            "0xedadc6f64383dc1df7c4b2d51b54225406d36b641f5e41bbc52a56612a8c6d14",
+            "0x00",
+            "0x00",
+        ],
+        [
+            "0x4bda12f684bda12f684bda12f684bda12f684bda12f684bda12f684b8e38e23c",
+            "0xc75e0c32d5cb7c0fa9d0a54b12a0a6d5647ab046d686da6fdffc90fc201d71a3",
+            "0x29a6194691f91a73715209ef6512e576722830a201be2018a765e85a9ecee931",
+            "0x2f684bda12f684bda12f684bda12f684bda12f684bda12f684bda12f38e38d84",
+        ],
+        [
+            "0xfffffffffffffffffffffffffffffffffffffffffffffffffffffffefffff93b",
+            "0x7a06534bb8bdb49fd5e9e6632722c2989467c1bfc8e8d978dfb425d2685c2573",
+            "0x6484aa716545ca2cf3a70c3fa8fe337e0a3d21162f0d6299a7bf8192bfd2a76f",
+            "0x00",
+        ],
+    ];
+    let mut k: [[<IsoSecp256k1 as CurveExt>::Base; 4]; 5] =
+        [[<IsoSecp256k1 as CurveExt>::Base::from_uniform_bytes(&[0; 64]); 4]; 5];
+    for i in 0..5 {
+        for j in 0..4 {
+            k[i][j] =
+                <IsoSecp256k1 as CurveExt>::Base::from_uniform_bytes(&hex_str_to_le_bytes(K[i][j]));
+        }
+    }
+
+    // convert to affine form
+    let (xp, yp, zp) = rp.jacobian_coordinates();
```

Comment:
Halo2 provides projective coordinates with a fully constant-time implementation.

Hashing-to-curve might be used for signing with secret keys so projective coordinates should be preferred.

Reply:
Can you give the reference of projective coordinates, if possible? @mratsim 

BTW, I think it is not much necessary to use projective coordinates(except for timing), since the output is the same.
What do you think? @mratsim

Reply:
I think I misunderstood. @mratsim 
The default EC point struct of curves in `halo2curves` is projective coordinates. 
Hence, I simplified the affine form conversion logic in https://github.com/privacy-scaling-explorations/halo2curves/pull/110/commits/161c661ed2bf098cc85a30d1e2702f2132fa1933.